osu.Game.Rulesets.Osu/Replays/OsuAutoGenerator.cs

```diff
@@ -255,6 +264,16 @@ private void moveToHitObject(OsuHitObject h, Vector2 targetPos, Easing easing)
                 for (double time = lastFrame.Time + GetFrameDelay(lastFrame.Time); time < h.StartTime; time += GetFrameDelay(time))
                 {
                     Vector2 currentPosition = Interpolation.ValueAt(time, lastPosition, targetPos, lastFrame.Time, h.StartTime, easing);
+
+                    //Adjust end frame postions
+                    var adjustingFrame = framesPendingAdjust.FirstOrDefault(frame => Precision.AlmostEquals(frame.Time, time, GetFrameDelay(time)));
+
+                    if (adjustingFrame != null)
+                    {
+                        adjustingFrame.Position = currentPosition;
+                        framesAfterAdjust.Add(adjustingFrame);
+                    }
```

Comment:
I have no idea what this is supposed to be doing and don't want to spend time on reverse-engineering your solution. The comments are useless because they say what's being done (which is redundant, since the code does so anyway), but they don't say why. The only thing I can about make out of this is that this conditional will always be true in the first iteration of this loop, which seems weird.

Can you clearly explain what is the idea behind this in detail?

Reply:
This searches for the current frames and see if there's any frame in framesPendingAdjust matches it in time.
If yes, we adjust that frame position to currentpostion, that's where the cursor should be at that time

Reply:
You're restating the code. Again.

Why this matching condition? When does it apply? What does the adjustment do?

Reply:
> You're restating the code. Again.
> 
> Why this matching condition? When does it apply? What does the adjustment do?
> 
> If you cannot explain the inner workings of this pull request's logic, you should not expect it to be merged.

Sorry, but I still don't know how to explain it, but will try as far as I can.

The problem this PR is trying to resolve is the key up frames uses wrong postions.

Auto should still be moving when it holds the key, so we add key up frames to `framesPendingAdjust`.

And when the moving to the next object `moveToHitObject()` gets called.

`currentPosition` is the position cursor should be at `time` so we see if there's any frame in `framesPendingAdjust` that has its time == `time`.